trio/_core/_ki.py

```diff
@@ -173,26 +183,34 @@ def wrapper(*args, **kwargs):
     return decorator
 
 
-enable_ki_protection: Callable[[F], F] = _ki_protection_decorator(True)
+enable_ki_protection: Callable[
+    [Callable[ArgsT, RetT]], Callable[ArgsT, RetT]
+] = _ki_protection_decorator(True)
 enable_ki_protection.__name__ = "enable_ki_protection"
 
-disable_ki_protection: Callable[[F], F] = _ki_protection_decorator(False)
+disable_ki_protection: Callable[
+    [Callable[ArgsT, RetT]], Callable[ArgsT, RetT]
+] = _ki_protection_decorator(False)
 disable_ki_protection.__name__ = "disable_ki_protection"
 
 
 @attr.s
 class KIManager:
     handler = attr.ib(default=None)
 
-    def install(self, deliver_cb, restrict_keyboard_interrupt_to_checkpoints):
+    def install(
+        self,
+        deliver_cb: Callable[[], object],
```

Comment:
I had typed `deliver_cb` as `Callable[[], None]` to indicate that the return value is unused. Ofc that means in exchange that you get a type error if you pass in a function that incidentally returns a value/is used elsewhere for it's return value/is not directly from the users code. Not sure which is superior/standard practice.

Reply:
[Best practice](https://typing.readthedocs.io/en/latest/source/best_practices.html#using-any-and-object) is to use `object` if you're ignoring the return type, since a function that does return something would be valid here.

Reply:
Good to know, thanks!
I don't think it's been in any of my PR's other than #2721